server.go

```diff
@@ -125,9 +135,39 @@ func main() {
 	s.Server = server
 
 	go server.Serve(listener)
-
 	trace.AuthRequest = func(req *http.Request) (any, sensitive bool) { return true, true }
 	log.Println("KMSServiceServer service started successfully.")
+	
+
+	go func() {
+		for {
+			s := <-sigChan
+			switch s {
+			case syscall.SIGHUP:
+				fmt.Println("hungup")
+			case syscall.SIGINT:
+				fmt.Println("Ctrl+c")
+				shutdown <- 0
+			case syscall.SIGTERM:
+				fmt.Println("force stop")
+				shutdown <- 0
+			case syscall.SIGQUIT:
+				fmt.Println("stop and core dump")
+				shutdown <- 0
+			default:
+				fmt.Println("Unknown signal.")
+				shutdown <- 1
+			}
+		}
+	}()
+
+	go func() {
+		code := <-shutdown
```

Comment:
why would we flag the process exist code as Error if it was forced to exit? it is not exactly an error